website_snippet_mass_mailing_partner/static/src/js/website_snippet_mass_mailing_partner.js

```diff
@@ -0,0 +1,57 @@
+/* © 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+"use strict";
+(function ($) {
+    openerp.website.snippet.animationRegistry.subscribe.include({
+        start: function(editable_mode) {
+            var self = this;
+            self._super(editable_mode);
+            if (self.$target.find(".js_subscribe_name:visible")) {
+                openerp.jsonRpc('/website_mass_mailing/is_subscriber', 'call', {
+                    list_id: this.$target.data('list-id'),
+                }).then(function (data) {
+                    self.$target.find('.js_subscribe_name:visible')
+                    .val(data.name)
+                    .attr(
+                        "disabled",
+                        Boolean(data.is_subscriber && data.name.length)
+                    );
+                });
+            }
+        },
+        on_click: function() {
```

Comment:
you break modularity with this override. `_super` is a mess that doesn't lend itself well for overriding, but still, don't break modularity. If necessary, write the name only after super is done with its stuff. This module anyways allows everyone to write every partner's name given the email address is known, which might not be a good idea in the first place.